js/glsl/mesh-fragment-lambert.glsl

```diff
@@ -0,0 +1,124 @@
+#extension GL_OES_standard_derivatives : enable
+#define LAMBERT
+
+uniform vec3 diffuse;
+uniform vec3 emissive;
+uniform float emissiveIntensity;
+uniform float opacity;
+
+varying vec3 vLightFront;
+
+#ifdef DOUBLE_SIDED
+
+	varying vec3 vLightBack;
+
+#endif
+
+#include <common>
+#include <packing>
+#include <dithering_pars_fragment>
+#include <color_pars_fragment>
+#include <uv_pars_fragment>
+#include <uv2_pars_fragment>
+#include <map_pars_fragment>
+#include <alphamap_pars_fragment>
+#include <aomap_pars_fragment>
+#include <lightmap_pars_fragment>
+#include <emissivemap_pars_fragment>
+#include <envmap_pars_fragment>
+#include <bsdfs>
+#include <lights_pars_begin>
+#include <fog_pars_fragment>
+#include <shadowmap_pars_fragment>
+#include <shadowmask_pars_fragment>
+#include <specularmap_pars_fragment>
+#include <logdepthbuf_pars_fragment>
+#include <clipping_planes_pars_fragment>
+
+//////////////////////////////////////////////////////////////////
+varying vec4 vertex_color;
+varying vec3 vertex_position;
+varying vec2 vertex_uv;
+
+#ifdef USE_TEXTURE
+    uniform sampler2D texture;
+    uniform sampler2D texture_previous;
+    uniform float animation_time_texture;
+#endif
+
+void main() 
+{
+	vec4 finalColor2 = vec4( 0.0, 0.0, 0.0, 1.0 );
+	
+#ifdef USE_RGB
+    finalColor2 = vec4( vertex_color.rgb, 1.0 );
```

Comment:
I think this color will be used for the lighting right? in the RGB pass it is important that we just render the vertex_color which represents the normalized xyz coordinates. These coordinates are used in the volume rendering pass as 'start' and 'end' coordinates.

Reply:
Also, I guess the finalColor2 is used in the threejs shader, do you have a pointer to some docs about this?

Reply:
The shader code must be optimized/refactored.